src/org/exist/xquery/parser/XQueryParser.java

```diff
@@ -84,7 +84,7 @@ public boolean foundErrors() {
 	}
 
 	public String getErrorMessage() {
-		StringBuilder buf= new StringBuilder();
+		StringBuffer buf= new StringBuffer();
```

Comment:
This was in the Antlr `XQuery.g` grammar which generates this Java file, I didn't change this previously. However, I just fixed it in a subsequent commit for you